chef/cookbooks/ceph/recipes/osd.rb

```diff
@@ -103,7 +110,15 @@
         Chef::Log.info("Ceph: Claimed #{d.name}")
         device = {}
         dev_name = d.name.gsub("/dev/", "")
-        if node["ceph"]["osd"]["journal_devices"].include?(d.name) || (node[:block_device][dev_name]["rotational"] == "0" && node["ceph"]["osd"]["use_ssd_for_journal"])
+        if node["ceph"]["osd"]["journal_devices"].include?(d.name) ||
+            (node[:block_device][dev_name]["rotational"] == "0" &&
+             node["ceph"]["osd"]["use_ssd_for_journal"] &&
+             node["ceph"]["disk_mode"] == "all")
```

Comment:
I wondered about doing that. As it stands now, you could apply once with `disk_mode == "first"`, have some disk allocated as an OSD, then later switch to `disk_mode == "all"`, reapply the proposal, and still get the `use_ssd_for_journal` behaviour when the remaining disks are deployed.

If I force `use_ssd_for_journal` to off if `disk_mode == "first"`, then that will remain off for that node even when potentially later switching to `disk_mode == "all"`. Does that matter? Would this be a loss? Or do we not care about this (admittedly bizarre) case?